Headers/Foundation/NSOrderedSet.h

```diff
@@ -0,0 +1,212 @@
+
+// Describing a set
+- (NSString *) description;
+- (NSString *) descriptionWithLocale: (NSLocale *)locale;
+- (NSString *) descriptionWithLocale: (NSLocale *)locale indent: (BOOL)flag;
+
+  // Convert to other types
+- (NSArray *) array;
+- (NSSet *) set;
+@end
+
+// Mutable Ordered Set
+@interface GS_GENERIC_CLASS(NSMutableOrderedSet, __covariant ElementT) : NSOrderedSet
+// Creating a Mutable Ordered Set
++ (instancetype)orderedSetWithCapacity: (NSUInteger)capacity;
+- (instancetype)initWithCapacity: (NSUInteger)capacity;
+- (instancetype) init;
+- (void)addObject:(GS_GENERIC_TYPE(ElementT))anObject;
+- (void)addObjects:(const GS_GENERIC_TYPE(ElementT)[])objects count:(NSUInteger)count;
+- (void)addObjectsFromArray:(GS_GENERIC_CLASS(NSArray, ElementT)*)otherArray;
+- (void)insertObject:(GS_GENERIC_TYPE(ElementT))object atIndex:(NSUInteger)index;
+- (void)setObject:(GS_GENERIC_TYPE(ElementT))object atIndexedSubscript:(NSUInteger)index;
+- (void)insertObjects:(GS_GENERIC_CLASS(NSArray, ElementT)*)array atIndexes:(NSIndexSet *)indexes;
+- (void)removeObject:(GS_GENERIC_TYPE(ElementT))object;
+- (void)removeObjectAtIndex:(NSUInteger)index;
+- (void)removeObjectsAtIndexes:(NSIndexSet *)indexes;
+- (void)removeObjectsInArray:(GS_GENERIC_CLASS(NSArray, ElementT)*)otherArray;
+- (void)removeObjectsInRange:(NSRange)range;
+- (void)removeAllObjects;
+- (void)replaceObjectAtIndex:(NSUInteger)index
+                  withObject:(GS_GENERIC_TYPE(ElementT))object;
+- (void) replaceObjectsAtIndexes: (NSIndexSet *)indexes
+                     withObjects: (GS_GENERIC_CLASS(NSArray, ElementT)*)objects;
+- (void) replaceObjectsInRange:(NSRange)range
+                   withObjects:(const GS_GENERIC_TYPE(ElementT)[])objects
+                         count: (NSUInteger)count;
+- (void)setObject:(GS_GENERIC_TYPE(ElementT))object atIndex:(NSUInteger)index;
+- (void)moveObjectsAtIndexes:(NSIndexSet *)indexes toIndex:(NSUInteger)index;
+- (void) exchangeObjectAtIndex:(NSUInteger)index withObjectAtIndex:(NSUInteger)otherIndex;
+- (void)filterUsingPredicate:(NSPredicate *)predicate;
+- (void) sortUsingDescriptors:(NSArray *)descriptors;
+- (void) sortUsingComparator: (NSComparator)comparator;
+- (void) sortWithOptions: (NSSortOptions)options
+         usingComparator: (NSComparator)comparator;  
+- (void) sortRange: (NSRange)range
+           options:(NSSortOptions)options
+   usingComparator: (NSComparator)comparator;
+- (void) intersectOrderedSet:(GS_GENERIC_CLASS(NSOrderedSet, ElementT)*)aSet;
+- (void) intersectSet:(GS_GENERIC_CLASS(NSSet, ElementT)*)aSet;
+- (void) minusOrderedSet:(GS_GENERIC_CLASS(NSOrderedSet, ElementT)*)aSet;
+- (void) minusSet:(GS_GENERIC_CLASS(NSSet, ElementT)*)aSet;
+- (void) unionOrderedSet:(GS_GENERIC_CLASS(NSOrderedSet, ElementT)*)aSet;
+- (void) unionSet:(GS_GENERIC_CLASS(NSSet, ElementT)*)aSet;
+- (instancetype) initWithCoder: (NSCoder *)coder;
+@end
+
+#endif
```

Comment:
This line should probably have a comment `/* _NSOrderedSet_h_GNUSTEP_BASE_INCLUDE */` like other public headers.

Also the file is missing the closing bracket for `extern "C"` opened at the top, i.e.:

```c
#if	defined(__cplusplus)
}
#endif
```